modnet/models.py

```diff
@@ -554,79 +566,904 @@ class OR only return the most probable class.
+
+def validate_model(train_data = None,
+                   val_data = None,
+                   targets = None,
+                   weights = None,
+                   num_classes=None,
+                   n_feat = 100,
+                   num_neurons = [[8],[8],[8],[8]],
+                   lr=0.1,
+                   batch_size = 64,
+                   epochs = 100,
+                   loss='mse',
+                   act = 'relu',
+                   out_act= 'linear',
+                   xscale='minmax',
+                   callbacks = [],
+                   preset_id = None,
+                   fold_id = None,
+                   verbose = 0,
+            ):
+
+    #deprecated
+    # data type can get messed up when passed to new process
+    #train_data.df_featurized = train_data.df_featurized.apply(pd.to_numeric)
+    #train_data.df_targets = train_data.df_targets.apply(pd.to_numeric)
+    #if val_data is not None:
+    #    val_data.df_featurized = val_data.df_featurized.apply(pd.to_numeric)
+    #    val_data.df_targets = val_data.df_targets.apply(pd.to_numeric)
+    #verbose=1
+
+    model = MODNetModel(
+        targets,
+        weights,
+        num_neurons=num_neurons,
+        n_feat=n_feat,
+        act=act,
+        out_act=out_act,
+        num_classes=num_classes
+    )
+
+    model.fit(
+        train_data,
+        lr = lr,
+        epochs = epochs,
+        batch_size = batch_size,
+        loss = loss,
+        xscale=xscale,
+        callbacks = callbacks,
+        verbose = verbose,
+        val_fraction = 0,
+        val_data = val_data,
+    )
+
+
+    learning_curve = model.history["val_loss"]
+
+    val_loss = model.evaluate(val_data)
+
+    #save model
+    model._make_picklable()
+
+    return val_loss, learning_curve, model, preset_id, fold_id
+
+
+def map_validate_model(kwargs):
+    return validate_model(**kwargs)
+
+
+#### Probabilistoc models####
+
+class Bayesian_MODNetModel(MODNetModel):
+    """Container class for the underlying Probabilistic Bayesian Neural Network, that handles
+    setting up the architecture, activations, training and learning curve. Only epistemic uncertainty is taken into account.
+
+    Attributes:
+        n_feat: The number of features used in the model.
+        weights: The relative loss weights for each target.
+        optimal_descriptors: The list of column names used
+            in training the model.
+        model: The `keras.model.Model` of the network itself.
+        target_names: The list of targets names that the model
+            was trained for.
+
+    """
+
+
+    def __init__(
+        self,
+        targets: List,
+        weights: Dict[str, float],
+        num_neurons=([64], [32], [16], [16]),
+        bayesian_layers=None,
+        prior=None,
+        posterior=None,
+        kl_weight=None,
+        num_classes: Optional[Dict[str, int]] = None,
+        n_feat: Optional[int] = 64,
+        act: str = "relu",
+        out_act: str = "linear",
+    ):
```

Comment:
It looks like this class only extends `MODNetModel` so it should be possible to re-use more of the init code and only override `build_model`

Reply:
Saved for next PR